### PR TITLE
Fix race

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

https://github.com/istio/istio/issues/51340

### CODE REVIEW

File: pilot/pkg/xds/ads.go [hunk 1]
```diff
@@ -627,28 +627,16 @@ func (s *DiscoveryServer) removeCon(conID string) {
 
 // nolint
 func (conn *Connection) NonceAcked(typeUrl string) string {
-	wr := conn.proxy.GetWatchedResource(typeUrl)
-	if wr != nil {
-		return wr.NonceAcked
```
Comment (wulianglongrd) [inline pilot/pkg/xds/ads.go:632]: `conn.proxy.GetWatchedResource` is protected by a lock. Here, we are just reading a string, so why is there a data race?

Comment (hzxuzhonghu) [reply pilot/pkg/xds/ads.go:632]: But wr.NonceAcked is not locked. 

Take a look at the original issue

Comment (wulianglongrd) [reply pilot/pkg/xds/ads.go:632]: Thanks, lgtm

Comment (wulianglongrd) [reply pilot/pkg/xds/ads.go:632]: I see other places calling `GetWatchedResource` and reading its return value (similar to the `NonceAcked` function), typically RDS/EDS reading `WatchedResource.ResourceNames`, which seems to have similar problems?

**Review by ramaraochavali — APPROVED** [review]

**Comment by howardjohn** [comment]
@hzxuzhonghu this never landed if you are interested in patching it up

**Pull request merged**